src/utils/PayloadValidation.ts

```diff
@@ -3,23 +3,31 @@ import ITechRecord from "../../@Types/ITechRecord";
 import {VEHICLE_TYPE, SEARCHCRITERIA} from "../assets/Enums";
 import Joi from "@hapi/joi";
 import {psvValidation} from "./PsvValidations";
+import {trlValidation} from "./TrlValidations";
+
+const checkIfTankOrBattery = (payload: ITechRecord) => {
+  let isTankOrBattery = false;
+  if (payload.adrDetails && payload.adrDetails.vehicleDetails && payload.adrDetails.vehicleDetails.type) {
+    const vehicleDetailsType = payload.adrDetails.vehicleDetails.type.toLowerCase();
+    if ((vehicleDetailsType.indexOf("battery") !== -1) || (vehicleDetailsType.indexOf("tank") !== -1)) {
+      isTankOrBattery = true;
+    }
```

Comment:
Why you don't just do a string comparison? Are you saving multiple separate types in the string `adrDetails.vehicleDetails.type` ?

Reply:
because vehicleDetails.type can be "rigid battery tank" or "semi trailer battery" I need to see if the word "tank" or "battery" is present in the vehicleDetails.type